mypy/semanal.py

```diff
@@ -6071,6 +6071,9 @@ def analyze_type_application_args(self, expr: IndexExpr) -> list[Type] | None:
                 return None
             types.append(analyzed)
 
+        analyzer = self.type_analyzer()
+        types = analyzer.check_unpacks_in_list(types)
+
```

Comment:
As a micro-optimization, I think it is better like this:
```suggestion
        if allow_unpack:
            types = self.type_analyzer().check_unpacks_in_list(types)
```